ci/lint.sh

```diff
@@ -5,5 +5,10 @@ if [ -x "$(command -v shellcheck)" ]; then
 else
 	echo "Warning: shellcheck is not installed, skipping shell scripts"
 fi
+incompatible_sed=$(! grep -RPn --color=always --exclude-dir=node_modules --exclude-dir=.git 'sed(?:\s-\w+)*\s-i(?!\S|\s"")')
```

Comment:
If you check the test cases, this particular regex pattern won't support all edge cases, would it be worth covering those edges? It will require more, and I probably have to consider awk to cover everything

Reply:
I think it's probably fine as it stands; we won't be trying to trick it, it'll just be people using `sed -i` rather than `sed -i.bak`